components/brave_wallet/browser/eth_topics_builder.cc

```diff
@@ -0,0 +1,39 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/eth_topics_builder.h"
+#include <utility>
+#include "brave/components/brave_wallet/browser/brave_wallet_utils.h"
+
+#include "brave/components/brave_wallet/common/hash_utils.h"
+#include "brave/components/brave_wallet/common/hex_utils.h"
+
+namespace brave_wallet {
+
+bool MakeAssetDiscoveryTopics(
```

Comment:
What's a topic in this context?

Reply:
Topics are a component of an Ethereum log.  This [post](https://medium.com/mycrypto/understanding-event-logs-on-the-ethereum-blockchain-f4ae7ba50378) explains it better than I could.  In this context, topics are effectively used as a query -- we are requesting all logs that match the topics made in this function.

Reply:
Ahh ok that makes more sense what's happening now. Thanks for the link.